### ISSUE TITLE
Danish date/time formats needs adjustment.

### ISSUE THREAD

**Issue by GustavBrock** [moment/moment#1428]
The Internationalization formats use the wrong separators and punction needs adjustment. It should read:

19-01-2014
19-1-2014
19. januar 2014
19. jan 2014
19. januar 2014 12.35
19. jan 2014 12.35
søndag 19. januar, 2014 12.35
søn 19. jan, 2014 12.35

**Comment by FredrikC** [comment]
The site referenced states the following (translated with google translate):
> Dates with slash hear especially at home in handwriting. You do not use sentence of dates with a slash, eg 2/10 2003. If you only have two digits of the year, one can possibly. put a hyphen between the month and year, eg 10/02 03 or 2/10-03.

It does not state that dd/MM/yyyy is used at all, it actually says:

> After the international dating system do you use all the order year, month and day. It is optional whether to use hyphens or not between the year and month and between month and day. Month and day are always written with two digits in the international system. The date October 2, 2003 signed by all the international system as follows: 20031002 or 2003-10-02. It is also possible to enter the two digit year: 031002 or. 03-10-02.
> 
> The order in traditional Danish date indication is day, month and year. You can choose to write the date with numbers alone: 10/02/2003 or 10/02/03 There is usually no need to add a zero at the days and months there is single-digit. One should therefore avoid writing 02/10/2003. The date written on the other hand never with only one digit, but with two or four digits.

I can't find any other good source of information since I do not speak Danish but I still suggest that we change to use '-'.

**Comment by GustavBrock** [comment]
I do speak Danish and I live in Denmark and see a lot of business correspondance.

In handwriting, anything is really accepted as long as it can be understood.

In business, the standard is dd-mm-yyyy or yyyy-mm-dd, where the first is a typical (old) domestic format widely used, and the latter is mostly seen at organisations operating world-wide wishing one format only.
A variation sometimes seen is replacing "-" with "." like dd.mm.yyyy, while using a "/" is very rarely used, though accepted.
Another variation is two digit years, but that was in the last Century where yy couldn't be taken for a date or month as is the case until 2032.
A variation almost never seen is the yyyymmdd format without separators.

To conclude: You are right - should only one separator be chosen, it should be "-". Nothing speaks against this.

**Comment by davidkallesen** [comment]
I also live in denmark and now have issues with the seperator '/'. Please reopen and change it from '/' to '-' for danish.

**Comment by tcarlsen** [comment]
Before this discussion takes further you all should really read why we ended up with using `/` ant not `-` here https://github.com/moment/moment/pull/1449.
I had changed it to `-` as you all able, but the original author of this translation wanted it back on `/`.

**Comment by GustavBrock** [comment]
I understand that the author's personal preference is "/" but, with all respect, that must have a lower priority than common usage. Also, I am not aware of any OS or programming language offering localized Danish date formats using a date separator other than "-" or ".". Here are three examples from official government sites:
    https://hjaelp.virk.dk/om-virk/virk-nyt/cvr-flytter-ind-paa-virk
    http://www.ft.dk/Aktuelt/Nyheder.aspx
    http://stm.dk/_a_1578.html
That should end the discussion.

**Comment by tcarlsen** [comment]
@GustavBrock as you will see in my PR back then I really don't care for that separator. 

so my advice: make PR with the changes and hope for a merge

**Comment by davidkallesen** [comment]
@GustavBrock - I was one of the backend developers on the ft.dk and the wrong date format was chosen based on a design companys work and not because of somebody actually had thought about it. And now we all think that the danish parliament at least use the correct format seperator. Shame but true :-(

**Comment by runemoennike** [comment]
I apologise for restarting this tired old discussion, but I'd like to point out that the resource linked to in #1449 does in fact **not support the argument that / and - can be used interchangeably** . Rather, in summary, it says:

1. The international system is acceptable and widely used: dd-mm-yy (or reversed yy-mm-dd, or without dashes at all.)
2. Traditionally, dates are written as d.m.yyyy or d.m.yy without zero padding.
3. Using forward slashes is prevalent in hand writing, but in the shape d/m yyyy (space between month and year, not slash).
4. Writing dates with dashes should be avoided in machine written texts.

In conclusion, and as many people have reported here, momentjs currently has the _wrong_ date format for Danish, and as such anyone wanting to use it with any of the correct formats have to adapt their own locale file.

**Comment by GustavBrock** [comment]
Thanks Rune. I thought this had been corrected. 
I cannot figure out what motive the original author can have for blocking this correction.

**Comment by KrisSiegel** [comment]
Has any movement been made here to correct the format for Danish?

**Comment by runemoennike** [comment]
Unfortunately it would seem not. I'm not sure what the process is here, should we create a new issue? Can someone reopen this one, perhaps @icambron?

As it stands now, the locale bundled with moment is useless for anyone wanting to use correct Danish date formatting as stated by the authority on the subject, which is a real shame. The work around is easy enough, of course, just call updateLocale with the correct settings, which is what we do.

**Comment by icambron** [comment]
*Sighs*. OK, so the way we do this is that we are willing to overturn the original author when doing so is indicated by an authoritative source. Usually that source ends up being CLDR because it's so easy to check. CLDR says Danish uses DD/MM/YYYY. I just checked to make sure no one had changed it since I last looked it up. CLDR could be wrong, but is sproget.dk the proper authority to overrule it? That's an honest question; it may well be, but how would I know? One of the government websites posted above as evidence uses two different formats and one actually uses DD/MM/YYYY!

And guys, come on, you can't just say "I speak Danish so you have to change this" or "I found a website on the Internet that agrees with me". Lots of people speak Danish and there are lots of websites. Similarly "this government official used this format on their website, so end of discussion" is plainly a bad way to go about this. We're not professional arbiters of language disputes; we maintain a date-time library. So please make this easy for us:

> According to [highly authoritative source] the short date format is [something]. It's highly authoritative because [some reason that will make sense to someone who doesn't speak Danish].

If someone can do that, I'd be happy to reopen this and accept a PR. I want the Danish date format to be right too!

**Comment by KrisSiegel** [comment]
That seems perfectly reasonable, @icambron. We were notified of this issue by someone in our translations group. I already reached out to find exactly what you're looking for (we also received similar guidance regarding Finnish).

**Comment by runemoennike** [comment]
I see your predicament, and I agree, arguments pertaining to nationality or websites are silly. I'm pretty sure I have made none of those, and if so, I apologise.

With that in mind, of the three websites linked above two now use the dot-notation, while the third uses dashes some places and dots in others. None of them use slashes, from the quick glance I just took.

As for the source: According to http://sproget.dk/om-sproget-dk/about-sproget.dk, sproget.dk is run as a collaboration between the Danish Ministry of Culture, the Danish Language Council and the Danish Society of Language and Literature. While Danish language is not technically a controlled language, the Danish Language Council is the closest thing to an authority on the subject. You can read their own statement in English here: https://dsn.dk/om-os/about-the-danish-language-council .

The language council runs a phone hotline as well as a mailbox where they answer questions on accepted usage of the Danish language. Answers that they find to be widely relevant are published on sproget.dk

Now to the final piece: http://sproget.dk/raad-og-regler/artikler-mv/svarbase/SV00000046 . As can be seen at the very top (unfortunately in Danish) this article is from the Danish Language Council's knowledgebase. Note how the date is written there, too hehe. To complete this chain of logic, I'll replicate my summary from above of what is said here:

1. The international system is acceptable and widely used: yyyy-mm-dd, yy-mm-dd, or without dashes at all.
2. Traditionally, dates are written as d.m.yyyy or d.m.yy without zero padding.
3. Using forward slashes is prevalent in hand writing, but in the shape d/m yyyy (space between month and year, not slash).
4. Writing dates with dashes should be avoided in machine written texts.

Note that nowhere at all does it even mention writing dates as dd/mm/yyyy, and about the dashes it says that d-mm-yy is usable in handwriting but poorly so in machine writing. Nowhere does it mention dd-mm-yyyy.

I'm sorry about the length of this, but I wanted to make the argument complete from end to end. If anyone has a more authoritative source than this, which says that dd/mm/yyyy or dd-mm-yyyy is acceptable or more correct, I'll happily bow down. As @icambron mentions, links to website using these formats is not an argument in itself, and should be written off to the heavy English influence on Danish culture. I imagine most developers in Denmark work with English sources daily, so it's no wonder this sneaks in everywhere.

**Comment by GustavBrock** [comment]
Thanks Rune. There is nothing more to add - except that it is unbelievable that this issue has survived for more than three years.

**Comment by icambron** [comment]
Thanks @runemoennike, that is helpful. Reopening.

**Comment by runemoennike** [comment]
You're welcome, and thank you!

I filed a bug with CLDR as well, to try and fix the problem at its source, so to speak. However, there was a similar bug report three years ago, which was denied based on talks with some people at Apple. I suspect their design wishes may have a strong influence, which is a bit sad. Let's see!

**Comment by icambron** [comment]
Fixed in #3945
